### PR TITLE
Dockerfile improvement

### PR DESCRIPTION
**Pull request by zhming0**
- Improves docker image security: use non root user, as numeric user id.
- Use fully qualified image url in `from` clause (so podman can build it too). 
- Use builder pattern so dockerfile is self contained now. 
- a `.dockerignore` file.

Thank you for making this software 🙂 .  I am enjoying using it.

### CODE REVIEW

**Comment by anderspitman** [comment]
Hi @zhming0. Thanks for the contribution! I probably won't merge this because I prefer the alpine images. They're much smaller in my experience. If you want to make similar improvements to the alpine image, they would be welcome, but don't feel any obligation to do so. I'm happy to keep this open so others can find your dockerfile. I'm sure some would find it useful.

**Comment by zhming0** [comment]
@anderspitman no problem at all. I changed it to Alpine now, please review 🙏🏿 . I love small image too 🙂.

FYI, in this case it's a 24MB vs 60MB difference.

**Comment by anderspitman** [comment]
Thanks @zhming0! I'll take a look soon. Since this Dockerfile is used for the production deployment I need to test and make sure it works.

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+# Copy the Go Modules manifests
+COPY go.mod go.sum ./
+# Download Go module dependencies
+RUN go mod download
+
+# Copy the source code into the container
+COPY . .
+
+# Those linker flag enables us to statically link sqlite
+RUN CGO_ENABLED=1 GOOS=linux go build -a -ldflags '-linkmode external -extldflags "-static"' -o obligator/obligator .
```
Comment (anderspitman) [inline docker/Dockerfile:15]: Can you use all the same flags as in the ./build.sh script? That should avoid the sqlite warnings. See this blog post for more background: https://www.arp242.net/static-go.html

Comment (anderspitman) [reply docker/Dockerfile:15]: Can you also strip the executable as well (may need to install binutils in the builder).

**Review by anderspitman — COMMENTED** [review]
I'm still seeing the docker image size as 44MB as opposed to 23MB for the previous Dockerfile. I'm assuming this is some sort of Docker image layer issue. Any idea how to fix it?

**Comment by anderspitman** [comment]
Turns out I was wrong. I'm not currently using this Dockerfile for production. I did a quick review and just have a couple requests.

File: docker/Dockerfile (hunk 1 above)
Comment (zhming0) [reply docker/Dockerfile]: I added in the `-tags sqlite_omit_load_extension,netgo` flag, now the build warnings are gone. Thanks for the link! 

Regarding the executable, I am not too sure which executable are you referring to, can you clarify it for me please?

Comment (anderspitman) [reply docker/Dockerfile]: For stripping, I'm referring to removing debug symbols from the obligator executable. Pretty sure the `strip` command is built into Debian, so you should be able to do something like:

```
RUN strip /obligator/obligator
```

Comment (zhming0) [reply docker/Dockerfile]: Ah I see. Done. Thanks for letting me know this trick 🙂 .

Comment (zhming0) [reply docker/Dockerfile]: @anderspitman do you mind taking another look? 🙏🏿

**Comment by anderspitman** [comment]
Sorry for the delay! Thanks again @zhming0

**Pull request merged**